app/src/main/java/com/example/florent/stateexperiment/ObservableViewModel.kt

```diff
@@ -10,7 +10,7 @@ abstract class ObservableViewModel<Input, Output> : ViewModel() {
     private val input: Subject<Input> = PublishSubject.create()
     private var output: Observable<Output>? = null
```

Comment:
since we have the input already why dont we initialise the pipeline here instead of doing a null check in caching() fun ?

Reply:
output is `input.compose(init)...`, you don't have `init` till you attach the presenter